test/testing/test_services.py

```diff
@@ -31,6 +35,23 @@ def reset_mock():
     handle_event.reset_mock()
 
 
+@pytest.yield_fixture
+def spawn_thread():
+
+    threads = []
+
+    def spawn(fn, *args):
+        threads.append(eventlet.spawn(fn, *args))
+
+    yield spawn
+
+    for gt in threads:
+        try:
+            gt.kill()
+        except:
```

Comment:
This is here to soak up any possible errors the greenlet threw. We're just tidying up -- do we need to limit it to `Exception`? I suppose if there was a `KeybaordInterrupt` in there we would want to let it bubble...